### PR TITLE
Run mount checks in parallel

### PR DESCRIPTION
**Pull request by Dr-Emann**
Also make prometheus an optional dependency

### CODE REVIEW

**Comment by acdha** [comment]
Thanks! Parallelism is something I was meaning to look into as a followup point release but since it's already done…

The only thing which jumped out during my first look over the code was that I don't love the `with_server` feature name. I'd been thinking about “with_prometheus” or “using_prometheus” but after looking over http://doc.crates.io/manifest.html#the-features-section I'm thinking that should just be “prometheus”.

**Comment by acdha** [comment]
Ah, Cargo won't allow features and dependencies to have the same name. I'll merge & rename it to “with_prometheus”.

**Comment by Dr-Emann** [comment]
Yea, that's what I originally tried to go with, but it wasn't possible. It seems like the case in which extra dependencies are required to support an optional dependency doesn't have the best of stories in cargo.

**Comment by acdha** [comment]
Yeah, it's not immediately obvious to me why this should be an error but I can appreciate the value of avoiding confusion and it's not like this is bad.

File: src/main.rs [hunk 1]
```diff
@@ -159,6 +159,45 @@ fn main() {
     }
 }
 
+#[cfg(not(feature = "with_server"))]
+fn push_to_server(_: &str, _: usize, _: usize) -> Result<(), &'static str> {
```
Comment (acdha) [inline src/main.rs]: I should have made a separate commit for this but a47611e2d452525ff60714a179b3e5f43c103ddb also tweaked the signature for the non-default version of push_to_server.

Comment (Dr-Emann) [reply src/main.rs]: Oops. Good catch.

**Pull request merged**